### PR TITLE
Setup codecov

### PR DESCRIPTION
**Pull request by Raynafs**
# Scope

_Please make sure to read the [Contribution Guidelines](https://github.com/droidconKE/droidconKE2023Android/blob/main/CONTRIBUTING.md)
and check that you understand and have followed it as best as possible Explain what your feature
does in a short paragraph._ please check the below boxes
- [ ] I have followed the coding conventions
- [ ] I have added/updated necessary tests
- [ ] I have tested the changes added on a physical device
- [ ] I have run `./codeAnalysis.sh` on linux/unix or `codeAnalysys.bat` on windows to make sure all lint/formatting checks have been done.

#283
_Declare any issues by typing `fixes #1` or `closes #1` for example so that the automation can kick
in when this is merged_

## Other testing QA Notes
_What have you tested specifically and what possible impacts/areas there are that may need retesting
by others._

Please add a screenshot (if necessary)

### CODE REVIEW

**Comment by Raynafs** [comment]
@chepsi still working on this. Don't understand why it automatically assigns reviewers

**Comment by kibettheophilus** [comment]
> @chepsi still working on this. Don't understand why it automatically assigns reviewers

the reviewers are requested automatically because of [Code Owners](https://github.com/droidconKE/droidconKeKotlin/blob/main/.github/CODEOWNERS), you can however convert the PR to draft if it is not ready for review 
Screenshot 2024-05-15 at 08 40 59

**Comment by Raynafs** [comment]
Thanks @kibettheophilus

**Comment by chepsi** [comment]
@Raynafs I have fixed the failing CI with my latest merge please rebase from develop

**Comment by Raynafs** [comment]
@chepsi @kibettheophilus This is ready, just needs to be tested in codecov.
If we can just activate this specific repository from here
https://app.codecov.io/gh/droidconke

File: .github/workflows/branch.yml [hunk 1]
```diff
@@ -89,4 +93,47 @@ jobs:
...
+        run: mkdir -p android
+
+      - name: Setup Gradle cache
+        uses: gradle/gradle-build-action@v2
+
+      - name: List root diretory contents
+        run: ls -la
+
+      - name: List android directory contents
+        run: ls -la android
```
Comment (chepsi) [inline .github/workflows/branch.yml:109]: Do we need all these steps?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -180,4 +181,9 @@ droidconke-android-application-firebase = { id = "droidconke.android.application
 droidconke-android-library-firebase = { id = "droidconke.android.library.firebase", version = "unspecified" }
 droidconke-android-application = { id = "droidconke.android.application", version = "unspecified" }
 kotlin-serialization = { id = "org.jetbrains.kotlin.plugin.serialization", version.ref = "kotlin" }
-droidconke-multiplatform = { id = "droidconke.multiplatform", version = "unspecified" }
```
Comment (chepsi) [inline gradle/libs.versions.toml:183]: why did we remove this?

**Review by chepsi — APPROVED** [review]

File: .github/workflows/branch.yml (hunk 1 above)
Comment (Raynafs) [reply .github/workflows/branch.yml:109]: Not really. I'll remove the root and android directory. I was using them to confirm gradle location

File: .github/workflows/branch.yml [hunk 3]
```diff
@@ -89,4 +93,41 @@ jobs:
...
+      - name: Ensure android directory exists
+        run: mkdir -p android
+
+      - name: Setup Gradle cache
+        uses: gradle/gradle-build-action@v2
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Run local tests for the combined coverage report (only API 30)
```
[the remark this answers is not in the document]
Comment (Raynafs) [reply .github/workflows/branch.yml:108]: Yeah lemme correct that

File: app/build.gradle.kts [hunk 4]
```diff
@@ -48,6 +49,7 @@ android {
                 getDefaultProguardFile("proguard-android-optimize.txt"),
                 "proguard-rules.pro"
             )
+            enableUnitTestCoverage = true
```
[the remark this answers is not in the document]
Comment (Raynafs) [reply app/build.gradle.kts]: yes sure lemme do so

**Review by kibettheophilus — APPROVED** [review]

**Pull request merged**